pandas/io/gbq.py

```diff
@@ -279,7 +279,7 @@ def _parse_data(schema, rows):
                                        field_type)
             page_array[row_num][col_num] = field_value
 
-    return DataFrame(page_array)
+    return DataFrame(page_array, columns=col_names)
```

Comment:
this should not be necessary, ``page_array`` is already a record array (this will just reindex it) (and copy it).